src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -482,7 +486,7 @@ private void insert(final SinkRecord record)
       }
 
       //ignore ingested files
-      if (record.kafkaOffset() > processedOffset.get())
+      if (record.kafkaOffset() >= pipeStatus.processedOffset.get())
```

Comment:
This seems like an unrelated change, how did you catch this ?

Reply:
When I was writing the pipeStatus comments, I found that processedOffset stores the offset of the most recent processed record. However, flushedOffset and commitOffset stores the offset of the most recent flushed/committed record offset + 1. So I think it's better to keep that consistent and let them all store last offset + 1.

Reply:
Also, this means that pipestatus is not just used for telemetry, right ? It contains actual state used during processing ?

Reply:
Correct, I moved processedOffset, flushedOffset, and committedOffset to the telemetry object.